src/components/oak.py

```diff
@@ -562,3 +602,103 @@ def __init__(self, method_name: str, details: str) -> None:
             f'Cannot invoke method "{method_name}" with current config. {details}'
         )
         super().__init__(self.message)
+
+
+class OakInstanceManager(Thread):
+    """
+    OakInstanceManager is a thread that manages every Oak instance.
+    It is responsible for orchestrating each Oak instance when one or more reconfigure requests are made.
+    """
+
+    def __init__(
+        self,
+        reqs: List[Tuple[Oak, ComponentConfig, Mapping[ResourceName, ResourceBase]]],
+    ) -> None:
+        super().__init__()
+        self.reconfig_reqs: List[
+            Tuple[Oak, ComponentConfig, Mapping[ResourceName, ResourceBase]]
+        ] = reqs
+        self.lock: Lock = Lock()
+        self.requests_waited: bool = False
+        self.oak_instances: List[Oak] = []
+        self.stop_event: asyncio.Event = asyncio.Event()
+        self.logger: Logger = getLogger("oak_manager")
+
+        self.logger.info("OakInstanceManager initialized with requests: %s", reqs)
+
+        self.loop = asyncio.new_event_loop()
+        self.loop.create_task(self.handle_reconfigures())
+
+    async def handle_reconfigures(self) -> None:
+        self.logger.info("Starting reconfiguration handler loop.")
+        while not self.stop_event.is_set():
+            await asyncio.sleep(0.1)
```

Comment:
Should we define sleep interval at the top of the file?

Reply:
Ah like as a const sure

Reply:
Done